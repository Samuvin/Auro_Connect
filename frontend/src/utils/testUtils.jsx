import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Create a custom render function that includes providers
export const renderWithProviders = (ui, options = {}) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  const AllTheProviders = ({ children }) => {
    return (
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

// Mock user data for testing
export const mockUser = {
  _id: '1',
  name: 'John Doe',
  username: 'johndoe',
  email: 'john@example.com',
  profilePicture: '/avatar.png',
  headline: 'Software Developer',
  connections: ['2', '3'],
  location: 'New York, NY'
};

// Mock post data for testing
export const mockPost = {
  _id: '1',
  content: 'This is a test post',
  author: mockUser,
  likes: ['2', '3'],
  comments: [
    {
      _id: '1',
      content: 'Great post!',
      user: mockUser,
      createdAt: new Date().toISOString()
    }
  ],
  image: 'https://example.com/image.jpg',
  createdAt: new Date().toISOString()
}; 