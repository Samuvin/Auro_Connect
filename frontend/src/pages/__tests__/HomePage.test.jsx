import { screen } from '@testing-library/react';
import HomePage from '../HomePage';
import { renderWithProviders } from '../../utils/testUtils';

// Mock the dependencies
jest.mock('../../lib/axios');

describe('HomePage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders homepage layout correctly', () => {
    renderWithProviders(<HomePage />);
    
    // Should render main content area
    expect(document.body).toBeInTheDocument();
  });

  test('contains main sections of the homepage', () => {
    renderWithProviders(<HomePage />);
    
    // Look for common homepage elements
    expect(document.body).toBeInTheDocument();
  });

  test('renders without authentication errors', () => {
    renderWithProviders(<HomePage />);
    
    // Should not show login prompts or error messages
    expect(screen.queryByText(/please log in/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/unauthorized/i)).not.toBeInTheDocument();
  });

  // Snapshot Tests
  test('matches snapshot for homepage layout', () => {
    const { container } = renderWithProviders(<HomePage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for homepage with mocked data', () => {
    // This will use the mocked data from setupTests.js
    const { container } = renderWithProviders(<HomePage />);
    expect(container.firstChild).toMatchSnapshot();
  });
}); 