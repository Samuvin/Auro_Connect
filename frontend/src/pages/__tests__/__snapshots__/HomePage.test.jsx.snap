// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HomePage matches snapshot for homepage layout 1`] = `
<div
  class="fixed bottom-10 right-10"
>
  <button
    class="relative p-4 bg-blue-500 text-white rounded-full shadow-lg flex items-center justify-center hover:bg-blue-700 transition-transform transform hover:scale-110"
  >
    <div
      class="absolute top-0 right-0 h-3 w-3 rounded-full bg-gray-400 border-2 border-white"
    />
    <svg
      fill="none"
      height="30"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 16V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l7-4A2 2 0 0 0 21 16z"
      />
      <polyline
        points="7.5 4.21 12 6.81 16.5 4.21"
      />
      <polyline
        points="7.5 19.79 7.5 14.6 3 12"
      />
      <polyline
        points="21 12 16.5 14.6 16.5 19.79"
      />
      <polyline
        points="3.27 6.96 12 12.01 20.73 6.96"
      />
      <line
        x1="12"
        x2="12"
        y1="22.08"
        y2="12"
      />
    </svg>
  </button>
</div>
`;

exports[`HomePage matches snapshot for homepage with mocked data 1`] = `
<div
  class="fixed bottom-10 right-10"
>
  <button
    class="relative p-4 bg-blue-500 text-white rounded-full shadow-lg flex items-center justify-center hover:bg-blue-700 transition-transform transform hover:scale-110"
  >
    <div
      class="absolute top-0 right-0 h-3 w-3 rounded-full bg-gray-400 border-2 border-white"
    />
    <svg
      fill="none"
      height="30"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 16V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l7-4A2 2 0 0 0 21 16z"
      />
      <polyline
        points="7.5 4.21 12 6.81 16.5 4.21"
      />
      <polyline
        points="7.5 19.79 7.5 14.6 3 12"
      />
      <polyline
        points="21 12 16.5 14.6 16.5 19.79"
      />
      <polyline
        points="3.27 6.96 12 12.01 20.73 6.96"
      />
      <line
        x1="12"
        x2="12"
        y1="22.08"
        y2="12"
      />
    </svg>
  </button>
</div>
`;
